src/components/common/dialog.tsx

```diff
@@ -0,0 +1,1550 @@
+                cached_result: undefined,
+                timeout_id: 0,
+                promise: undefined,
+                round_id: undefined,
+            };
+        return this.#validation_state[path];
+    }
+
+    /* Calling #set_validation_state_result is the final thing that
+       should happen when validating a given path. It will install the
+       result in the cache and then call #set_validation.
+     */
+
+    #set_validation_state_result(
+        state: DialogValidationState,
+        val: unknown,
+        result: string | undefined,
+    ) {
+        state.cached_value = val;
+        state.cached_result = result;
+        state.timeout_id = 0;
+        state.promise = undefined;
+        state.round_id = undefined;
+        this.#set_validation(state.path, result);
+    }
+
+    /* The first thing should be of course to probe that cache.  If we
+       get a hit, it is used immediately to call #set_validation.
+
+       In that case, the DialogValidationState is also made part of
+       the current round since any asynchronous validation that is
+       currently running is still relevant. See below for more about
+       that.
+     */
+
+    #probe_validation_state_cache(state: DialogValidationState, val: unknown): boolean {
+        if (Object.is(state.cached_value, val)) {
+            state.round_id = this.#get_current_validation_round_id();
+            debug("cache hit", state.path, state.cached_result);
+            this.#set_validation(state.path, state.cached_result);
+            return true;
+        } else
+            return false;
+    }
+
+    /* And in fact, _validate_value does exactly those two things.
+     */
+
+    _validate_value(path: string, val: unknown, func: () => string | undefined): void {
+        const state = this.#get_validation_state(path);
+        if (!this.#probe_validation_state_cache(state, val)) {
+            const result = func();
+            debug("sync validate", state.path, result);
+            this.#set_validation_state_result(state, val, result);
+        }
+    }
+
+    /* Now asynchronous validation.
+
+       Each call to #trigger_validation starts a new "validation
+       round" and a DialogValidationState keeps track to which round
+       it applies to.  This matters of course for asynchronous
+       validation: If async validation for a given path was started in
+       one round, and then the next round happens but the path is no
+       longer enumerated by the validation callback (i.e., its value
+       is no longer relevant for the dialog), then this asynchronous
+       validation should have no effect.
+
+       We use the #validation object as the round identifier, since it
+       is created fresh by each call to #trigger_validation.
+     */
+
+    #get_current_validation_round_id(): unknown {
+        return this.#validation;
+    }
+
+    #is_current_validation_round_id(id: unknown): boolean {
+        return Object.is(id, this.#validation);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2435-0ef97292-20251218-063518-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)